### DISCUSSION TITLE
How to implement input hint on ui.input

### DISCUSSION THREAD

**Discussion by wasabismile** [zauberzeug/nicegui#3617] [Q&A] [answered]
### Question

I tried to set the autocomplete attribute and found that it can automatically complete a single word.
If I continue to enter another word on ui.input, there is no automatic prompt.
For example, if I enter apple, it can be automatically completed, and then continue to enter goodyear, there is no prompt.
What should I do to achieve continuous automatic completion. After entering a word, enter a specific separator, such as '/', and continue to enter the next word, it can also be automatically completed
```python
import re
from nicegui import ui
names = ['Google', 'Facebook', 'Twitter', 'Apple', 'Oracle', 'Goodyear']
ui.input(autocomplete=names)
ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @wasabismile,

You can update the autocomplete suggestions on every value change. Here is an example replacing the last word with fruits:

```py
fruits = ['Apple', 'Banana', 'Cherry', 'Date', 'Elderberry', 'Fig', 'Grape']

def handle_change():
    words = input_.value.split()
    input_.set_autocomplete([' '.join(words[:-1] + [fruit]) for fruit in fruits])

input_ = ui.input(on_change=handle_change)
```

<img width="167" alt="Screenshot 2024-08-28 at 07 29 13" src="https://github.com/user-attachments/assets/82e55d8d-1149-4796-b4a3-ebffa6f70aaa">

**Comment by wasabismile** [comment]
@falkoschindler  Thank you very much, you always help me a lot
